### PR TITLE
Default to dry-run operations

### PR DESCRIPTION
**Pull request by rappizs**
Fixes https://github.com/uwu-tools/gh-jira-issue-sync/issues/36 although I'm not sure if this documentation is enough or not

**Dev notes:**
~- added missing fields to the update logic~
~- made all fields optional so empty fields won't be added to the user's config file~
- refactored dry run logic, now dry run is the default mode
- updated docs

### CODE REVIEW

File: README.md
Comment (justaugustus) [inline README.md:1]: (First, flagging that this is related to https://github.com/uwu-tools/gh-jira-issue-sync/issues/36.)

I wrestle with this every time I deal with dry-run operations.
It's been a bit since I've touched this code, but I believe the reason that dry-run is not included as an option in this config is because of the default values.

If you include `dry-run` as a boolean field and do not set it, then it is initialized as the `boolean` type's zero value (which is `false`).
In other words, this means dry-runs must be explicitly requested (and the behavior you actually want is the other way around).

In tools like `peribolos`, we instead prefer a `--confirm` flag/option, which must be explicitly set to `true` for a production run: https://github.com/uwu-tools/peribolos/blob/c84a844017865578458a626222eacbaefa01817c/README.md?plain=1#L226-L228

File: internal/config/config.go [hunk 1]
```diff
@@ -272,18 +272,20 @@ func (c *Config) SetJiraToken(token *oauth1.Token) {
...
-	JiraSecret  string        `json:"jira-secret" mapstructure:"jira-secret"`
-	JiraKey     string        `json:"jira-private-key-path" mapstructure:"jira-private-key-path"`
-	JiraCKey    string        `json:"jira-consumer-key" mapstructure:"jira-consumer-key"`
-	RepoName    string        `json:"repo-name" mapstructure:"repo-name"`
-	JiraURI     string        `json:"jira-uri" mapstructure:"jira-uri"`
-	JiraProject string        `json:"jira-project" mapstructure:"jira-project"`
-	Since       string        `json:"since" mapstructure:"since"`
-	Timeout     time.Duration `json:"timeout" mapstructure:"timeout"`
+	LogLevel    string        `json:"log-level,omitempty" mapstructure:"log-level"`
+	DryRun      string        `json:"dry-run,omitempty" mapstructure:"dry-run"`
```
Comment (justaugustus) [inline internal/config/config.go:276]: See comment above about dry-run operations.
Also, the README suggests that this field is a boolean, but it is defined as a string here.
Prefer booleans in scenarios where string interpretation can lead to unexpected results e.g., `configFile.DryRun = "is-this-interpreted-as-true-or-false?"`

File: internal/config/config.go [hunk 2]
```diff
@@ -272,18 +272,20 @@ func (c *Config) SetJiraToken(token *oauth1.Token) {
...
+	GithubToken string        `json:"github-token,omitempty" mapstructure:"github-token"`
+	JiraUser    string        `json:"jira-user,omitempty" mapstructure:"jira-user"`
+	JiraPass    string        `json:"jira-pass,omitempty" mapstructure:"jira-pass"`
```
Comment (justaugustus) [inline internal/config/config.go:279]: Are we sure we want to include passwords in plaintext?
ref: https://github.com/uwu-tools/gh-jira-issue-sync/issues/34

**Review by justaugustus — CHANGES_REQUESTED** [review]
@rappizs -- Thanks for submitting this!

I think this PR may be conflating two features/bug fixes:
- making sure the config file remains consistent: https://github.com/uwu-tools/gh-jira-issue-sync/issues/43
- documenting/enabling dry-run operations: https://github.com/uwu-tools/gh-jira-issue-sync/issues/36

I'd suggest splitting the "pass by value" / "pass by reference" out into a separate PR, as that looks to be mergeable by itself.

File: internal/config/config.go (hunk 1 above)
Comment (rappizs) [reply internal/config/config.go:276]: The type is my mistake, should be bool of course

File: README.md
Comment (rappizs) [reply README.md:1]: I totally agree with the --confirm approach but for this refactoring we should create a separate issue in my opinion.

File: internal/config/config.go (hunk 2 above)
Comment (rappizs) [reply internal/config/config.go:279]: This change is only about updating the user's config file, which the tool does after every run, it has nothing to do with the format of the password. It's always provided by the user as plain text but if we are going to remove username+password as an auth option then this shouldn't be a problem.

File: README.md
Comment (rappizs) [reply README.md:1]: Agreed, but it's already introduced, dry-run is a supported config option ATM it's just not documented and it always gets deleted from the config after every run because it's missing from this struct, which we use to update the config file. But it's handled and used by the business logic with or without this PR. Should I add the dry-run refactoring logic to this PR as well?

File: internal/config/config.go (hunk 2 above)
Comment (justaugustus) [reply internal/config/config.go:279]: Fair point. Let's keep that for now then!

File: README.md
Comment (justaugustus) [reply README.md:1]: Yeah, I think the reason I explicitly left it out of the config struct was you don't want to run into a scenario where you're doing testing and the config file is set to do a production run. The way it exists today forces the user to explicitly set the desired behavior from the command line.

I'm fine with a refactor to include it in the struct here, so long as the default behavior is `confirm: false`.

Comment (rappizs) [reply README.md:1]: > The way it exists today forces the user to explicitly set the desired behavior

Unfortunately not, if the user doesn't specifically set the dry-run property to true in the config file, then it will be a production run. Since it's not a mandatory config field, it will be set to false by default as you mentioned above. But this behavior is completely independent from this PR, this is how it works currently, production run is the default, dry-run is an extra option.

**Review by justaugustus — APPROVED** [review]
